### PR TITLE
Detective Equipment Beacon 

### PR DESCRIPTION
**Pull request by pireamaineach**
Detective closet very cluttered, best way to clear up space without ruining the aesthetic of the office.
Adds the VNP uniform, also removes the noir suit coat

vnp
 
## Changelog
:cl:
add: Add the detective equipment beacon and VNP alternative uniform.
/:cl:

### CODE REVIEW

**Comment by pireamaineach** [comment]
Probably going to be redoing the Vichy uniform either today or tommorow.

**Comment by Twaticus** [comment]
equipment beacon is a really good idea but theres no reason to remove the skirts or the coat. just because you dont use it doesnt mean other people dont. besides the grey detective skirt is objectively the best skirt wtf?

**Comment by pireamaineach** [comment]
I literally made the coat.

Skirt man bad. Content bloat with little substance.

**Comment by LemonInTheDark** [comment]
> Also removes the skirts and the noir suit coat nobody uses.

I use the noir suit coat.... Please don't cramp my style.

**Comment by pireamaineach** [comment]
The suit coat or the trench coat?

**Comment by LemonInTheDark** [comment]
Suit coat

**Comment by pireamaineach** [comment]
I mean you could always just wear one of the other suits in the game, rather than creating another outfit for the difference of 1 mildly different jacket.

**Comment by pireamaineach** [comment]
Also you could wait until I add the cloak which is objectively the most epique police fashion known to man

**Comment by LemonInTheDark** [comment]
Based, the promised addition will cure all ills

**Comment by pireamaineach** [comment]
Compiling to see how it looks ingame. Might be done.

**Comment by ATH1909** [comment]
Move the gun out of the beacon so that non-detectives can still loot it from the locker.

**Comment by pireamaineach** [comment]
@ATH1909 I thought about this, the beacon is not role restricted.

**Comment by pireamaineach** [comment]
It's the same as the curator's beacon. There's also 2 different guns dependant on loadout.

**Comment by ATH1909** [comment]
Also, you didn't mention anywhere that the vpndet revolver has different stats to the normal detective revolve-

Oh, you just did. Still, you should probably mention that in the OP/changelog.

**Comment by pireamaineach** [comment]
It has the same stats.

**Comment by pireamaineach** [comment]
Different ammo type but they do the same thing.

**Comment by pireamaineach** [comment]
It's just not a .38.

**Comment by ATH1909** [comment]
Or add a conversion kit or something to the locker so that detectives who want to use the best detective revolver aren't shoehorned into sticking with one specific style.

**Comment by ATH1909** [comment]
Oh, they do?

**Comment by pireamaineach** [comment]
Oui.

**Comment by ATH1909** [comment]
Ah, you're right.

Why not just make it a subtype of the .38 revolver, then? Why give it its own special ammo type?

**Comment by pireamaineach** [comment]
Because making an MAS 1892 that fires .38 would make me want to shoot myself.

**Comment by pireamaineach** [comment]
I am probably being dramatic but you get the point.

**Comment by pireamaineach** [comment]
I am preparing some pictures before I commit, moment.

**Comment by pireamaineach** [comment]
@LemonInTheDark Cloak pics up.

**Comment by SweptWasTaken** [comment]
inciting votes on discord?

KINDA CRINGE NGL

**Comment by necromanceranne** [comment]
Cloaks <3

**Comment by pireamaineach** [comment]
Why is Travis fuck

**Comment by necromanceranne** [comment]
Check the runtimes. There is an issue with your ammo design.

**Comment by pireamaineach** [comment]
I like how in favour of suggesting changes so they can see what they want from this P.R. people are opting for the, "Don't do this. Downvoted."

**Comment by LemonInTheDark** [comment]
Blessed outfit, thank you for being so accommodating, it's real nice of you.

**Comment by pireamaineach** [comment]
I will now make big iron

**Comment by HomocidlePancakes** [comment]
Time to mimedet.

File: code/game/objects/structures/crates_lockers/closets/secure/security.dm [hunk 1]
```diff
@@ -181,29 +181,22 @@
 
 /obj/structure/closet/secure_closet/detective/PopulateContents()
 	..()
+	new /obj/item/choice_beacon/detective(src)
 	new /obj/item/clothing/under/rank/security/detective(src)
-	new /obj/item/clothing/under/rank/security/detective/skirt(src)
```
Comment (tralezab) [inline code/game/objects/structures/crates_lockers/closets/secure/security.dm]: i haven't checked the comments yet but i bet people are mad at this

File: code/game/objects/structures/crates_lockers/closets/secure/security.dm [hunk 2]
```diff
@@ -181,29 +181,22 @@
...
 /obj/structure/closet/secure_closet/detective/PopulateContents()
 	..()
+	new /obj/item/choice_beacon/detective(src)
 	new /obj/item/clothing/under/rank/security/detective(src)
-	new /obj/item/clothing/under/rank/security/detective/skirt(src)
 	new /obj/item/clothing/suit/det_suit(src)
 	new /obj/item/clothing/head/fedora/det_hat(src)
 	new /obj/item/clothing/gloves/color/black(src)
-	new /obj/item/clothing/under/rank/security/detective/grey(src)
-	new /obj/item/clothing/under/rank/security/detective/grey/skirt(src)
```
Comment (tralezab) [inline code/game/objects/structures/crates_lockers/closets/secure/security.dm]: or in layman's terms.., these must be included. you can't remove this. this must be in the game

File: code/modules/projectiles/guns/ballistic/revolver.dm [hunk 3]
```diff
@@ -133,6 +130,14 @@
 			to_chat(user, "<span class='notice'>You remove the modifications on [src]. Now it will fire .38 rounds.</span>")
 	return TRUE
 
+/obj/item/gun/ballistic/revolver/detective/modele1892
```
Comment (tralezab) [inline code/modules/projectiles/guns/ballistic/revolver.dm]: otherwise, lots of cool stuff going on here!

**Review by tralezab — COMMENTED** [review]
trouble is brewing

**Comment by cacogen** [comment]
It's been cluttered for ages but a better fix would be a vendor, and atomising this PR so that any removals are done in a different one.

**Comment by pireamaineach** [comment]
Venders clash with the theming of the detective's office.

**Comment by ATH1909** [comment]
Are detective equipment beacons useable by anyone, or does this PR basically kill detective locker raiding?

Also, even if you don't make the MAS 1892 fire .38 bullets, you should still make the bullets that it fires a subtype of .38 bullets, in case .38 bullets get buffed, nerfed, etc. in the future.

Also, can the MAS 1892 fire the .38 bullet variants that are available, such as icebox rounds and tracking rounds? If we can't make the MAS 1892 fire .38 bullets, can we make the normal detective revolver fire 8mm rounds instead?

File: code/modules/projectiles/projectile/bullets/revolver.dm [hunk 4]
```diff
@@ -55,6 +55,10 @@
 		var/mob/living/M = target
 		M.adjust_bodytemperature(((100-blocked)/100)*(temperature - M.bodytemperature))
 
+/obj/projectile/bullet/fo8
+	name = "8mm French Ordnance bullet"
```
Comment (ATH1909) [inline code/modules/projectiles/projectile/bullets/revolver.dm]: *Space French

File: code/modules/projectiles/guns/ballistic/revolver.dm [hunk 5]
```diff
@@ -133,6 +130,14 @@
...
+	name = "\improper Modèle 1892 revolver"
+	desc = "A french smokeless powder revolver produced by Manufacture d'armes de Saint-Étienne. Fires a special 8mm cartridge."
```
Comment (ATH1909) [inline code/modules/projectiles/guns/ballistic/revolver.dm]: "french" should be capitalized (and replaced with "Space French" anyway, no?)

Comment (Fikou) [reply code/modules/projectiles/guns/ballistic/revolver.dm]: "1892 revolver"

**Comment by pireamaineach** [comment]
Locker raiding still an option, sub-type probably valid.

**Comment by Victor239** [comment]
The laceups have also been removed I see. Why sneakers or jackboots only?

**Comment by pireamaineach** [comment]
Because it's an outfit why would it have 2 pairs of shoes

**Comment by Victor239** [comment]
It should be kept in the closet is my point.

**Comment by pireamaineach** [comment]
I don't see a point. Just go to dorms.

**Comment by pireamaineach** [comment]
Skirt gang is yet to suggest a way to incorporate skirts.

**Comment by SkeletalElite** [comment]
> Skirt gang is yet to suggest a way to incorporate skirts.

Add skirts as options in the beacon?
Forget the beacon and add a vendor to the dets room like the chaps?

**Comment by Mickyan** [comment]
The closet is still unreasonably cluttered if less so, even lawyers get a vendor there's no reason to do it like this

**Comment by pireamaineach** [comment]
Completely wrong there's like 8 things in there now

**Comment by pireamaineach** [comment]
Plus the alternative gun still requires a beacon.

**Comment by Mickyan** [comment]
Also to complain about content bloat while adding another alternate outfit is kind of rich

**Comment by pireamaineach** [comment]
It's an alternative outfit that doesn't just reduce to, "same thing but skirt!!!!"

**Comment by Mickyan** [comment]
I count 15 which is a bit higher than 8

**Comment by pireamaineach** [comment]
Hyperbole. It's not absurd to sort through anymore, which I find more important. It also doesn't class with the design of the detective's office.

I'd like to avoid adding entirely new boxes for the dumb fucking skirts but tacking them into the standard boxes seems additionally worse.

**Comment by pireamaineach** [comment]
I'll create 1 extra for the noir suit since the det can spawn with a hard-worn skirt.

**Comment by pireamaineach** [comment]
I have never seen anyone use either of these, they're hardly unique items, the fact that people throw this much of a fuss about a creatively bankrupt and useless item is beyond me.

**Comment by pireamaineach** [comment]
It's also 13 not 15 but straws.

**Comment by pireamaineach** [comment]
HEWWO SKIRTS BACK NOW

**Comment by pireamaineach** [comment]
@Mickyan also you got your shoes

File: code/modules/clothing/under/jobs/security.dm [hunk 6]
```diff
@@ -133,6 +133,13 @@
 	can_adjust = FALSE
 	fitted = FEMALE_UNIFORM_TOP
 
+/obj/item/clothing/under/rank/security/detective/vpndet
+	name = "police nationale patrolman uniform"
+	desc = "Maréchal, nous voilà !"
```
Comment (ATH1909) [inline code/modules/clothing/under/jobs/security.dm]: is there supposed to be a space before the exclamation mark?

**Comment by ATH1909** [comment]
Why not just add "Classic Noir (Skirt)" and "L.A. Noir (Skirt)" as options for the detective beacon, alongside the normal "Classic Noir" and "L.A. Noir" options?

**Comment by pireamaineach** [comment]
Because grouping them into two different boxes instead of just adding one line is lame.

File: code/modules/clothing/under/jobs/security.dm (hunk 6 above)
Comment (pireamaineach) [reply code/modules/clothing/under/jobs/security.dm]: Yes

File: code/modules/clothing/shoes/miscellaneous.dm [hunk 7]
```diff
@@ -126,6 +126,12 @@
 	permeability_coefficient = 0.05 //Thick soles, and covers the ankle
 	pocket_storage_component_path = /datum/component/storage/concrete/pockets/shoes
 
+/obj/item/clothing/shoes/jackboots/vpndet
+	name = "police nationale service shoes"
+	desc = "Walk in the footsteps of those who came before."
```
Comment (Fikou) [inline code/modules/clothing/shoes/miscellaneous.dm]: 😳

**Comment by ExcessiveUseOfCobblestone** [comment]
so now if someone steals your gun they also steal all your clothing options?

😭

File: code/modules/clothing/shoes/miscellaneous.dm (hunk 7 above)
Comment (pireamaineach) [reply code/modules/clothing/shoes/miscellaneous.dm]: what

**Comment by pireamaineach** [comment]
Why are people still upvoting a comment and downvoting the P.R. for an already fixed issue ?

**Comment by optimumtact** [comment]
Adding a beacon will not escape my bloat complaints, knock it off.

Every job doesn't need 300 alternate outfits, add it to the costume crate.

**Comment by pireamaineach** [comment]
@optimumtact but oranges the original PR removed 2 outfits and added one why wouldn't you just give your opinion on that

**Comment by optimumtact** [comment]
your ideological war on skirts is not my concern.

**Comment by pireamaineach** [comment]
"This is bloat because the detective has too many uniforms."

"You are not allowed to remove uniforms."

**Comment by optimumtact** [comment]
your ideological war on skirts is not my concern.

**Comment by cacogen** [comment]
@pireamaineach

>Venders clash with the theming of the detective's office.

i don't think it matters but a early 20th century vending machine (e.g. those in fallout or bioshock) would suffice
